flocker/node/agents/blockdevice.py

```diff
@@ -1356,8 +1313,9 @@ def is_existing_block_device(dataset_id, path):
             volumes=volumes,
             devices=devices,
             system_mounts=system_mounts,
-            devices_with_filesystems=[device for device in devices.values()
-                                      if _has_filesystem(device)],
+            devices_with_filesystems=[
+                device for device in devices.values()
+                if self.block_device_manager.has_filesystem(device)],
```

Comment:
:grey_question: I wonder if this should be on the interface: given these `devices` return the ones with filesystems.